### DISCUSSION TITLE
How to create a table on database over controller

### DISCUSSION THREAD

**Discussion by stripedbug** [adonisjs/core#3873] [Help]
Hi

I need to create a table and if possible a model for the table from a controller . 
I know i can do that with some third party node pack but is there a built in way to do that in adonis.

**Comment by RomainLanz** [comment]
Hey @stripedbug! 👋🏻 

May I ask why would you want to do that? In my opinion, it looks like a [XY Problem](https://xyproblem.info/).
But if you really need to do that, you could use `ace` as a `child_process` to generate those.

**Reply by stripedbug** [reply]
What i'm trying to do is to create a seperate table dedicated for the result of some calculations. In my application, data i m collecting is dynamic, meaning the input areas are user generated. After that those input field data will be calculated with also custom formulas which are user generated too.

So rather than calculating all data all at once every time i want to view a report, i m creating  a dedicated table for every form groups and populate it as the users submit the forms. 

What i do now is create a table with mysql package and interact with it via database facade. I only wanted to know if there is a more straight forward way to do it

**Reply by RomainLanz** [reply]
It seems that you need to cache the result of heavy computation.
Any reason not to store those results in a cache (like Redis) instead of keeping them in the database?

**Reply by stripedbug** [reply]
> It seems that you need to cache the result of heavy computation. Any reason not to store those results in a cache (like Redis) instead of keeping them in the database?

The results will gradually get bigger as users keep submitting and i dont think cache is a solid way to store those data. Also the user submission process might take months so redis or any kind of cache method is not ideal

**Reply by andrew-connell** [reply]
Why don't you just use Postgresql and have just one table with a jsonb column for user specified data. Or a NoSQL db like Mongo.
Creating a lot of tables is a terrible idea, you're definitely overcomplicating things.

**Reply by stripedbug** [reply]
Thank u for the advice. Original question was "how to create a table on controller"

do u have an advice on that matter?

**Reply by andrew-connell** [reply]
No - as @RomainLanz says, this is definitely an XY problem
